kobo/rcS

```diff
@@ -62,8 +69,19 @@ fi
 /bin/mknod /dev/null c 1 3
 /bin/chmod 666 /dev/null
 
-/sbin/udevd --daemon
-/sbin/udevadm trigger
+# udev doesn't create all /dev/ entries on some platforms (e.g. Nia),
```

Comment:
Thanks for a quick review! This is what the supplied init script on the Nia is doing, so I suspect we won't fully understand it without knowing the hardware details. I'll pull out C++ changes into a separate commit, but for here our options are either to use this code, change it to uncompress the .tgz unconditionally if it's available, or use mknod to create the device tree if some required /dev/ entries are missing. My worry with unconditionally using .tgz is that it's different from what Kobo's init is doing, which is known to work on all platforms. 

Basically what happened is that sometime between 6.8 and 7 rcS switched to calling udev instead of manually creating /dev/ entires. On the Nia this does not work, and installing XCSoar 7 bricks it.

Reply:
I poked around a bit more and and added documentation to what this is doing. Note that the same code is present in the original Kobo boot script, and in [TopHat](https://github.com/rdunning0823/tophat/blob/TopHat_master/kobo/rcS.tophat#L67), so it should allow XCSoar to boot on more platforms without impacting others. 

Let me know if you prefer to use one of the other approaches I outlined above, though personally I'd prefer to follow what is proven to work in existing boot scripts.

Reply:
I'd prefer not to apply this ugly kludge to devices which don't need it. Keep the scope of bad code as small as possible.

Reply:
It already works this way, with additional comments in https://github.com/XCSoar/XCSoar/pull/634/commits/31e922b8f2eb6c35dc62595db4f5f0d715100510. If you don't like the approach taken by the device manufacturer, you can either roll back  https://github.com/XCSoar/XCSoar/commit/2de78d4aa22cee2913db0031befa0f51019e3d2b, pick from multiple options I repeatedly provided in here, or continue bricking Kobo Nia (and likely later) devices with XCSoar 7.

Reply:
> It already works this way

No, it doesn't. You changed the behavior of all devices, not just the Kobo Nia.

Reply:
The behavior is conditional on the platform. We're going in circles here, if you don't like this approach either pick one of the alternatives, as I keep asking, or make a concrete suggestion. This has stopped being productive a few iterations ago.

Reply:
> The behavior is conditional on the platform.

Yes, but that only appears to contradict with my statement "You changed the behavior of all devices, not just the Kobo Nia".

> We're going in circles here

True, because I see code changes where I don't want any changes, but you don't seem to see those changes in your own code. If you want to support the Kobo Nia and take responsibility for that platform, do that, but don't change behavior for all the other Kobos. (Unless you want to be the responsible maintainer for all of them, which I'd very much welcome.)

Reply:
> True, because I see code changes where I don't want any changes,

Then pick one of the options I've repeatedly suggested instead of ignoring them entirely, just like you did here. This entire discussion is you saying "I don't like it" to the approach taken by the device manufacturer, with no suggestions of your own nor any acknowledgement of multiple options I keep offering.

Reply:
Maybe this wasn't clear enough:
> I'd prefer not to apply this ugly kludge to devices which don't need it. Keep the scope of bad code as small as possible.

This means: go ahead with bad code, but limit it to devices that really really need it; i.e. only Kobo Nia. Do not change anything for all other devices.